common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/IdentityState.scala

```diff
@@ -1,41 +1,39 @@
 package uk.ac.wellcome.models.work.internal
 
-sealed trait IdentityState[+T]
-
-sealed trait Unminted[+T] extends IdentityState[T] {
-  val agent: T
-  def withAgent[A](f: T => A): Unminted[A]
+sealed trait IdState {
```

Comment:
Some comments explaining how you expect these traits/case classes to be used would be helpful, e.g. looking at this file I'm not clear on the distinction between "Minted" and "Unminted".

Reply:
:+1: none of these traits / case classes are actually new, just that the structure has been modified a bit. Definitely agree now is a good time for some documentation, so will add some in.

Reply:
Have added some doc strings to this file